contrib/storage-hive/core/src/test/java/org/apache/drill/exec/fn/hive/TestInbuiltHiveUDFs.java

```diff
@@ -114,33 +115,33 @@ public void testRand() throws Exception {
             .go();
   }
 
-  @Test //DRILL-4868
+  @Test //DRILL-4868 & DRILL-2326
   public void testEmbeddedHiveFunctionCall() throws Exception {
-    // TODO(DRILL-2326) temporary until we fix the scalar replacement bug for this case
-    final OptionValue srOption = QueryTestUtil.setupScalarReplacementOption(bits[0], ClassTransformer.ScalarReplacementOption.TRY);
+    String query =
+        "SELECT convert_from(unhex(key2), 'INT_BE') as intkey \n" +
+        "FROM cp.`functions/conv/conv.json`";
+
+    List<String> compilers = Arrays.asList(ClassCompilerSelector.CompilerPolicy.JANINO.name(),
+        ClassCompilerSelector.CompilerPolicy.JDK.name());
 
     try {
-      final String[] queries = {
-          "SELECT convert_from(unhex(key2), 'INT_BE') as intkey \n" +
-              "FROM cp.`functions/conv/conv.json`",
-      };
+      setSessionOption(ExecConstants.SCALAR_REPLACEMENT_OPTION, ClassTransformer.ScalarReplacementOption.ON.name());
+      setSessionOption(ClassCompilerSelector.JAVA_COMPILER_DEBUG_OPTION, false);
```

Comment:
Looks like `ClassCompilerSelector.JAVA_COMPILER_DEBUG_OPTION` can be unchanged. So then for the debug level this test case can log the generated code (according the Javadoc for the `ClassCompilerSelector`)